Client/core/CDiscordRichPresence.h

```diff
@@ -0,0 +1,43 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        core/CDiscordRichPresence.h
+ *  PURPOSE:     Header file for discord rich presence
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+#pragma once
+
+#include <core/CDiscordInterface.h>
+#include <string>
+
+class CDiscordRichPresence : public CDiscordInterface
+{
+public:
+    CDiscordRichPresence();
+    ~CDiscordRichPresence();
+
+    // void SetApplicationId();
+    void UpdatePresence();
+    void SetPresenceState(const char* szState);
+    void SetPresenceDetails(const char* szDetails);
+    void SetPresenceStartTimestamp(const unsigned long ulStart);
+    // void SetPresenceTimestamp();
+    // void SetPresenceImage();
+    // void SetPresenceText();
+
+private:
+    const char* m_strDiscordAppId;
+    const char* m_strDiscordAppAsset;
+    const char* m_strDiscordAppAssetSmall;
+    const char* m_strDiscordAppAssetText;
+    const char* m_strDiscordAppAssetSmallText;
+
+    const char*   m_strDiscordAppCurrentId;
+    const char*   m_strDiscordAppState;
+    const char*   m_strDiscordAppDetails;
```

Comment:
Please don't store pointers to characters if you can not guarantee the memory behind these pointer will remain always valid for the entire lifetime. This is a trap for other developers if they are not aware of this design choice. Use `std::string`.